jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -362,6 +362,32 @@ private static PrintStream createConsole(FileDescriptor desc) {
 	/*[ENDIF] Sidecar18-SE-OpenJ9 */
 	/*[ENDIF] JAVA_SPEC_VERSION >= 11 */
 
+	static void initJFR() {
+		String jfr = com.ibm.oti.vm.VM.getjfrCMDLineOption();
+		if (jfr != null) {
+			try {
+				Class<?> dcmdClass = Class.forName("jdk.jfr.internal.dcmd.DCmdStart");
+				Constructor<?> constructor = dcmdClass.getDeclaredConstructor();
+				constructor.setAccessible(true);
+				Object dcmdInstance = constructor.newInstance();
+
+				Method executeMethod = dcmdClass.getSuperclass().getDeclaredMethod("execute", String.class, String.class, char.class);
+				executeMethod.setAccessible(true);
+
+				String[] results = (String []) executeMethod.invoke(dcmdInstance, "internal", jfr, ',');
+				if (results != null) {
+					for (String result : results) {
```

Comment:
Is this for debugging. You can add a system property like `-Denable.j9internal.jfr.debug` to enable it

Reply:
@tajila  - Below is the sample output in hotspot, when JFR is enabled

```
> java -XX:StartFlightRecording InflaterTest2
[0.941s][info][jfr,startup] Started recording 1. No limit specified, using maxsize=250MB as default.
[0.941s][info][jfr,startup] 
[0.941s][info][jfr,startup] Use jcmd 901021 JFR.dump name=1 filename=FILEPATH to copy recording data to file.
```

Should I still add this as part of debug property?

Reply:
Okay, I wasn't sure if hotpot was doing this. Are they piping it to stdout or stderr?

Reply:
@tajila - I thought that those messages will be printed as is to console, but I was wrong. They are actually part of logging.

So in hotspot, if the jfr is enabled through commandline and if the logging is enabled for JFR then it is piped to stdout. By default, JFR logging `info` is `ON` and can be tuned using `-Xlog:jfr+startup=error/off`